pkg/metadata/inventories_collector.go

```
@@ -19,34 +17,32 @@ const (
 	maxSendInterval = 10 * time.Minute
 )
 
-type schedulerInterface interface {
-	AddCollector(name string, interval time.Duration) error
-	TriggerAndResetCollectorTimer(name string, delay time.Duration)
+type inventoriesCollector struct {
+	ac   inventories.AutoConfigInterface
+	coll inventories.CollectorInterface
+	sc   *Scheduler
 }
 
-type autoConfigInterface interface {
-	GetLoadedConfigs() map[string]integration.Config
-}
+var (
+	expvarPayload func() interface{}
+)
 
-type collectorInterface interface {
-	GetAllInstanceIDs(checkName string) []check.ID
-}
+func (c inventoriesCollector) createPayload() (*inventories.Payload, error) {
+	hostname, err := util.GetHostname()
+	if err != nil {
+		return nil, fmt.Errorf("unable to submit inventories metadata payload, no hostname: %s", err)
+	}
 
-type inventoriesCollector struct {
-	ac   autoConfigInterface
-	coll collectorInterface
-	sc   schedulerInterface
+	return inventories.GetPayload(hostname, c.ac, c.coll), nil
 }
 
 // Send collects the data needed and submits the payload
 func (c inventoriesCollector) Send(s *serializer.Serializer) error {
-	hostname, err := util.GetHostname()
+	payload, err := c.createPayload()
 	if err != nil {
-		return fmt.Errorf("unable to submit inventories metadata payload, no hostname: %s", err)
+		return nil
```

we don't want to propagate the error?

---

Good catch